README.md

```diff
@@ -91,113 +94,220 @@ The front end of the integration is now complete.
+    secondScenarioBuilder.clientSDKID = @"YOUR_SECOND_CLIENT_SDK_ID";
+    secondScenarioBuilder.scenarioID = @"YOUR_SECOND_SCENARIO_ID";
+    secondScenarioBuilder.callbackBackendURL = [NSURL URLWithString:@"YOUR_CALLBACK_URL"];
+    YTBScenario *secondScenario = [secondScenarioBuilder create:&error];
+    [YotiSDK addScenario: secondScenario];
+
+    return YES;
+```
+then in your viewController class inside your button IBAction function call the function  : 
+
+```swift
+
+public static func startScenario(for useCaseID: String, with delegate: YotiSDKDelegate) throws 
 ```
-And then you can add the scenario to Yoti SDK with
+Next, go to the view controller containing the YotiButton outlet. On its IBAction, call the sdk method signature in swift or signature in objective-c.
 
 Swift:
+
 ```swift
-YotiSDK.add(scenario: scenario)
+
+@IBAction func yotiButtonDidTouchUpInside(_ sender: YotiButton) {
+    guard let useCaseID = sender.useCaseID else {
+        return
+    }
+    do {
+        UIApplication.shared.isNetworkActivityIndicatorVisible = true
+        try YotiSDK.startScenario(for: useCaseID, with: self)
+    } catch {
+        // Handle error here
+    }
+}
 ```
+
+Objective-C: 
+
+```objective-c
+
+- (IBAction)buttonDidTouchUpInside:(YotiButton*)sender {
+    NSString* useCaseID = sender.useCaseID;
+    NSError* error = nil;
+
+    if (![useCaseID isEqual:@""]) {
+        [YotiSDK startScenarioForUseCaseID:useCaseID withDelegate:self error:&error];
+
+        if (error != nil) {
+            NSLog(@"error : %@", error.description);
+        }
+    }
+}
+```
+
+In Swift your ViewController class should comply to YotiSDKDelegate and to BackendDelegate in order to get the callbacks.
+
+```swift
+
+extension ViewController: YotiSDKDelegate {
+    func yotiSDKDidFail(for useCaseID: String, with error: Error) {
+    	// handle here the error related to the failing of retrieving a usecaseID and a token
+    }
+
+    func yotiSDKDidSucceed(for useCaseID: String, baseURL: URL?, token: String?, url: URL?) {
+        // Handle here the success of the opening of Yoti app for example by requesting a profile from the backend like below
+        // Get the specific scenario by calling  
+        let scenario = YotiSDK.scenario(for: useCaseID)
+        // request the backend to get the profile linked to a specific scenario by passing the token returned and self as delegate for a call back
+        YotiSDK.callbackBackend(scenario: scenario!, token: token!, with: self)
+    }
+
+    func yotiSDKDidOpenYotiApp() {
+        // Handle specific behaviour if needed when the Yoti App didOpen
+    }
+}
+```
+
+In Objective-C, your viewController should comply to  YTBSDKDelegate and YTBBackendDelegate like this:
+
+```objective-c
+
+@interface ViewController () <YTBSDKDelegate, YTBBackendDelegate>
+```
+
+Note that complying to YTBBackendDelegate is optional. BackendDelegate is its name in swift, YTBBackendDelegate namespace is for Objective-C
```

Comment:
You will need to explain why BackendDelegate is optional and in which conditions you will want to not implement it